### PR TITLE
Add representative segmentation sweep and protocol validation

### PR DESCRIPTION
**Pull request by calebrob6**
## Summary

- add a resumable multi-GPU runner for the representative segmentation matrix
- commit all 624 supported results from 640 candidate combinations, plus a configuration fingerprint
- add a cross-model protocol-sensitivity study covering 168 controlled comparisons
- preserve complete pretrained normalization chains for channel-adapted TorchGeo models
- resize temporal/multimodal GeoBench V2 images and masks consistently
- record the actual segmentation learning rate, batch size, and per-epoch validation history

## Representative matrix

- 8 representative backbones
- 10 GeoBench V2 segmentation datasets
- 4 probe heads: linear, conv-block, FPN, and DPT
- RGB and all-band inputs
- 10 epochs at 224 px
- 16 model/input combinations explicitly excluded because they violate pretrained input-normalization contracts

Results: `results/segmentation_representative_rgb_all_224_10ep.csv`

## Is the fixed evaluation protocol fair across models?

The expanded study holds the frozen-feature evaluation constant and varies only the probe optimization protocol:

- **7 backbone families:** DINOv3 ConvNeXt-Large, ResNet-50, EfficientNet-B0, DenseNet-121, MaxViT-Tiny, ScaleMAE-Large, and Satlas SwinV2-B
- **4 dataset difficulty levels:** Burn Scars, CloudSEN12, DynamicEarthNet, and PASTIS
- **6 schedules:** the 10-epoch baseline; 30/50 epochs; LR 3e-4/1e-3/3e-3; cosine versus constant LR
- **Uniform probe batch 8** for every model
- **FPN head, frozen features, 224 px, single seed**
- Protocols are ranked using **final validation mIoU**, never test mIoU

This produces **168 comparisons**. Relative to the default 10 epochs, LR 1e-3, cosine schedule:

| Variant | Mean final-val delta | Bootstrap 95% CI |
|---|---:|---:|
| 30 epochs, LR 3e-4, cosine | -0.0006 | [-0.0081, 0.0065] |
| 30 epochs, LR 1e-3, cosine | -0.0030 | [-0.0148, 0.0061] |
| 30 epochs, LR 3e-3, cosine | -0.0050 | [-0.0151, 0.0025] |
| 50 epochs, LR 1e-3, cosine | -0.0076 | [-0.0241, 0.0048] |
| 30 epochs, LR 1e-3, constant | **-0.0097** | **[-0.0185, -0.0025]** |

### Conclusions

- The **10-epoch, LR 1e-3, cosine baseline has the highest mean final validation mIoU** across all 28 model/dataset pairs.
- No longer or alternative-LR schedule produces a statistically supported aggregate improvement.
- Selecting one schedule per model by validation changes mean test mIoU by only **+0.0012**, so model rankings are not being driven by a poorly chosen shared schedule.
- Longer schedules frequently overfit DynamicEarthNet and PASTIS; constant LR is significantly worse.
- The default is therefore a defensible, compute-efficient common criterion across CNN, transformer, ImageNet-pretrained, and geospatial-pretrained backbones.

Artifacts:

- `scripts/run_segmentation_protocol_study.py` — resumable seven-model study runner with strict provenance/output validation
- `scripts/analyze_segmentation_protocol_study.py` — validation-based analysis and bootstrap confidence intervals
- `results/segmentation_protocol_study.csv` — all 168 comparisons
- `results/segmentation_protocol_study_summary.csv` — aggregate validation deltas and CIs
- `results/segmentation_protocol_study_selected.csv` — validation-selected schedule per model

## External sanity check

The resulting mIoUs are broadly consistent with the closest official GeoBench V2 reference: frozen DOFA-Large + UNet with HPO and five seeds. SpaceNet2 is nearly identical (0.855 here versus 0.848 official), and most datasets are within expected model/probe variation. PASTIS remains flagged as structurally difficult for a single-image frozen probe because the original task is temporal.

Official reference: https://github.com/The-AI-Alliance/GEO-Bench-2-Leaderboard/tree/main/results/80100fc6-dc1a-4514-b946-341157eaf816

## Validation

- `pytest tests/test_torchgeo_models.py tests/test_main_segmentation_fast.py tests/test_silent_bug_regressions.py tests/test_geobench_v2_datasets.py --no-cov -q`
- `pytest tests/test_segmentation.py --no-cov -q` — 40 passed
- `ruff check` and `ruff format --check` on changed Python files
- representative runner resume check against the complete 624-row CSV
- protocol runner/analyzer dry run, strict metadata/output validation, and 168/168 completed comparisons

### CODE REVIEW

**Review by isaaccorley — APPROVED** [review]
lgtm
